js/app.mjs

```diff
@@ -26,114 +23,115 @@ async function generateImage() {
+  if (!event.clipboardData.types.includes('Files')) {
     event.preventDefault();
-    var text = event.clipboardData.getData("text/plain");
-    document.execCommand("insertHTML", false, text);
+    const text = event.clipboardData.getData('text/plain');
+    document.execCommand('insertHTML', false, text);
   }
-})
-
+});
 
 /**
  * Event listeners on input fields
  */
+const ID = {
+  HANDWRITING: '#handwriting-font',
+  PEN_COLOR: '#ink-color',
+  FONT_SIZE: '#font-size',
+  TOP_PADDING: '#top-padding',
+  WORD_SPACING: '#word-spacing',
+  FONT_FILE: '#font-file',
+  TOGGLE_MARGIN: '#paper-margin-toggle',
+};
 
-document.querySelector('select#handwriting-font')
-  .addEventListener('change', e => textareaEl.style.fontFamily = e.target.value)
-
-document.querySelector('select#ink-color')
-  .addEventListener('change', e => {
-    textareaEl.style.color = e.target.value;
-    setInkColor(e.target.value);
-  })
-
-document.querySelector('input#font-size')
-  .addEventListener('change', e => textareaEl.style.fontSize = e.target.value + 'pt')
-
-document.querySelector('input#top-padding')
-  .addEventListener('change', e => textareaEl.style.paddingTop = e.target.value + 'px')
-
-document.querySelector('input#word-spacing')
-  .addEventListener('change', e => textareaEl.style.wordSpacing = e.target.value + 'px')
-
-document.querySelector('#font-file')
-  .addEventListener('change', e => {
-    addFontFromFile(e.target.files[0])
-  })
-
-document.querySelector('#paper-margin-toggle')
-  .addEventListener('change', e => 
-    document.querySelector('.page').classList.toggle('margined-page')
-  )
-
-document.querySelector('button#draw-diagram-button')
-  .addEventListener('click', toggleDrawCanvas)
-
-document.querySelector('.draw-container .close-button')
-  .addEventListener('click', toggleDrawCanvas)
-
-// Generate image on form submit
-document.querySelector('form#generate-image-form')
-  .addEventListener('submit', e => {
-    e.preventDefault(); // prevents reloading
-    generateImage();
-  })
-
-
-/***
- * Mostly the code nobody has to care about
-*/
+/**
+ *
+ * @param {string} attrib
+ * @param {any} value
+ */
+const setStyle = (attrib, v) => {
+  textareaEl.style[attrib] = v;
+};
 
-function smoothlyScrollTo(hashval) {
-  let target = document.querySelector(hashval)
-  target.scrollIntoView({
-    behavior: 'smooth',
-    block: 'start'
-  })
-  history.pushState(null, null, hashval)
-}
+/**
+ *
+ * @param {ChangeEvent} e
+ */
+const handleToolEvents = (e) => {
+  e.preventDefault();
+
+  const id = '#' + e.target.getAttribute('id');
+  switch (id) {
+    case ID.HANDWRITING:
+      setStyle('fontFamily', e.target.value);
+      break;
+    case ID.FONT_SIZE:
+      setStyle('fontSize', e.target.value + 'pt');
+      break;
+    case ID.TOP_PADDING:
+      setStyle('paddingTop', e.target.value + 'px');
+      break;
+    case ID.WORD_SPACING:
+      setStyle('wordSpacing', e.target.value + 'px');
+      break;
+    case ID.FONT_FILE:
```

Comment:
Isn't this more complicated to maintain than having `document.querySelector().addEventListener()`? Let's say I want to add new eventListener, as of what I understand now, 
1. I will have to add the selector to `ID` object
2. add a case in this switch statement
3. put functions inside the case
4. Make sure if I `break` it.

Plus this is because I know this repository. If a new person wants to add an event handler, he will have to go and see how these functions work and figure out everything.

Reply:
I would argue that this approach is more manageable and easier to maintain, every event can be looked up easily. and its cleaner too in my opinion.

also not that it really matters but it also prevents adding lot of EventListeners to the document, its just one event listener to manage all the change events.

although your points are also valid :D

*robost code is like redux... makes boilerplate but can be beneficial in long run*

Reply:
I also had one another approach to handle these events then i ditched it, have a look at it

```js
const EVENT_MAP = {
  '#handwriting-font': {
    on: 'change',
    action: (e) => setStyle('fontFamily', e.target.value),
  },
  '#font-size': {
    on: 'change',
    action: (e) => setStyle('fontSize', e.target.value + 'pt'),
  },
  '#word-spacing': {
    on: 'change',
    action: (e) => setStyle('wordSpacing', e.target.value + 'px'),
  },
};

for (const event in EVENT_MAP) {
  document
    .querySelector(event)
    .addEventListener(EVENT_MAP[event].on, EVENT_MAP[event].action);
}
```

Reply:
> robost code is like redux... makes boilerplate but can be beneficial in the long run

True! But we can always scale up as we need. In the past, I've made lots of projects complicated than they are supposed to be because I tried to have robust code.



> I also had one another approach to handle these events then i ditched it, have a look at it
> 
> ```js
> const EVENT_MAP = {
>   '#handwriting-font': {
>     on: 'change',
>     action: (e) => setStyle('fontFamily', e.target.value),
>   },
>   '#font-size': {
>     on: 'change',
>     action: (e) => setStyle('fontSize', e.target.value + 'pt'),
>   },
>   '#word-spacing': {
>     on: 'change',
>     action: (e) => setStyle('wordSpacing', e.target.value + 'px'),
>   },
> };
> 
> for (const event in EVENT_MAP) {
>   document
>     .querySelector(event)
>     .addEventListener(EVENT_MAP[event].on, EVENT_MAP[event].action);
> }
> ```

This looks pretty cool! I guess we can go with this :D any particular reason for ditching this approach?

Reply:
> This looks pretty cool! I guess we can go with this :D any particular reason for ditching this approach?

Okay i'll push the changes with this method. there was no reason for ditching this method other than i liked the previous approach more. xD